test/e2e/cache/create_enable_for_private_registry_delete.go

```diff
@@ -0,0 +1,336 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package cache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/test/framework"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"golang.org/x/crypto/bcrypt"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener-extension-registry-cache/pkg/apis/registry/v1alpha3"
+	"github.com/gardener/gardener-extension-registry-cache/test/common"
+	"github.com/gardener/gardener-extension-registry-cache/test/e2e"
+)
+
+const (
+	nginx1240             = "nginx:1.24.0"
+	registry300beta1Image = "europe-docker.pkg.dev/gardener-project/releases/3rd/registry:3.0.0-beta.1"
+	upstreamConfigYAML    = `version: 0.1
+log:
+  fields:
+    service: registry
+storage:
+  filesystem:
+    rootdirectory: /var/lib/registry
+auth:
+  htpasswd:
+    realm: basic-realm
+    path: /var/lib/password/htpasswd
+http:
+  addr: :5000
+health:
+  storagedriver:
+    enabled: true
+    interval: 10s
+    threshold: 3
+`
+)
+
+var _ = Describe("Registry Cache Extension Tests", Label("cache"), func() {
+	parentCtx := context.Background()
+
+	f := e2e.DefaultShootCreationFramework()
+	f.Shoot = e2e.DefaultShoot("e2e-cache-pr")
+
+	var (
+		password          string
+		encryptedPassword []byte
+		secret            *corev1.Secret
+	)
+
+	BeforeEach(func() {
+		ctx, cancel := context.WithTimeout(parentCtx, 2*time.Minute)
+		defer cancel()
+
+		// Prepare htpasswd
+		var err error
+		password, err = utils.GenerateRandomString(32)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(password)).To(Equal(32))
+
+		// Create Secret in the Project namespace
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "ro-upstream-secret",
+				Namespace: f.ProjectNamespace,
+			},
+			Immutable: ptr.To(true),
+			Type:      corev1.SecretTypeOpaque,
+			Data: map[string][]byte{
+				"username": []byte("admin"),
+				"password": []byte(password),
+			},
+		}
+		Expect(f.GardenClient.Client().Create(ctx, secret)).To(Succeed())
+
+		encryptedPassword, err = bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
Cosmetic nit for consistency with the rest of the project: 

```suggestion
		Expect(err).NotTo(HaveOccurred())
```